tests/queries/0_stateless/02403_date_time_narrowing.sql

```diff
@@ -0,0 +1,51 @@
+SELECT toDate(toDate32('1930-01-01', 'UTC'), 'UTC'),
+       toDate(toDate32('2150-01-01', 'UTC'), 'UTC');
+
+SELECT toDate(toDateTime64('1930-01-01 12:12:12.12', 3, 'UTC'), 'UTC'),
+       toDate(toDateTime64('2150-01-01 12:12:12.12', 3, 'UTC'), 'UTC');
+
+SELECT toDateTime(toDateTime64('1930-01-01 12:12:12.12', 3, 'UTC'), 'UTC'),
+       toDateTime(toDateTime64('2150-01-01 12:12:12.12', 3, 'UTC'), 'UTC');
+
+SELECT toDateTime(toDate32('1930-01-01', 'UTC'), 'UTC'),
+       toDateTime(toDate32('2150-01-01', 'UTC'), 'UTC');
+
+SELECT toDateTime(toDate('2140-01-01', 'UTC'), 'UTC');
+
+SELECT 'toStartOfDay';
+SELECT toStartOfDay(toDate('2141-01-01', 'UTC'), 'UTC'),
+       toStartOfDay(toDate('1941-01-01', 'UTC'), 'UTC'),
```

Comment:
> '2141-01-01' is a valid date.
Indeed [it is](https://clickhouse.com/docs/ru/sql-reference/data-types/date/), but it is not a valid [DateTime](https://clickhouse.com/docs/ru/sql-reference/data-types/datetime/) -- this can cause a lot of confusion (maybe it already did for someone)